instruction-error/src/lib.rs

```diff
@@ -245,7 +208,6 @@ pub enum InstructionError {
 #[cfg(feature = "std")]
 impl std::error::Error for InstructionError {}
```

Comment:
Is there a reason to avoid using `core::error::Error` when deriving these? We should be able to essentially remove `std` with that, and only enable it for frozen-abi builds

Reply:
you'd be be bumping the msrv to 1.81, which is higher than solana-program's msrv

Reply:
Ah right, we can do that in the future then, no need to do it here